src/org/loklak/harvester/BaseScraper.java

```diff
@@ -0,0 +1,121 @@
+import org.json.JSONException;
+import org.loklak.objects.ProviderType;
+import org.loklak.objects.SourceType;
+import org.loklak.objects.Timeline;
+import org.loklak.objects.QueryEntry;
+import java.util.HashMap;
+import javax.servlet.ServletException;
+import javax.servlet.http.Cookie;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import org.loklak.tools.storage.JSONObjectWithDefault;
+import org.json.JSONObject;
+import org.loklak.data.DAO;
+import org.loklak.http.ClientConnection;
+import org.loklak.server.*;
+import java.io.IOException;
+import org.loklak.http.RemoteAccess;
+import org.loklak.tools.UTF8;
+
+/**
+ * @author vibhcool (Vibhor Verma)
+ * @version 0.1
+ * @since 07.06.2017
+ *
+ * This is Base-Interface for Search-Scrappers.
+ */
+public abstract class BaseScraper extends AbstractAPIHandler {
+
+    // a time stamp that is given by loklak upon the arrival of the post which is the current local time
+    //TODO: check if UTC time needed
+    protected String scraperName;
+    protected String html;
+    protected String url;
+    protected String baseUrl;
+    protected String midUrl = "";
+    protected String query;
+    protected SourceType source_type; // where did the message come from
+    protected ProviderType provider_type;  // who created the message
+    //TODO: dummy variable, add datastructure for filter, type_of_posts, location, etc
+    protected String extra = "";
+    //TODO: setup Timeline for Post
+    protected final Timeline.Order order = Timeline.parseOrder("timestamp");
+
+    @Override
+	public JSONObject serviceImpl(Query call, HttpServletResponse response, Authorization rights,
+			JSONObjectWithDefault permissions) throws APIException {
+        this.query = call.get("query", "");
+        
+        //TODO: add different extra paramenters. this is dummy variable
+        this.extra = call.get("extra", "");
+        //TODO: to be implemented to use Timeline
+        //return getData().toJSON;
+        return this.getData();
+	}
+
+    protected abstract Map<?, ?> getExtra(String _extra);
+   
+//    public Timeline getData() {
+    public Post getData() {
+        ClientConnection connection;
+        BufferedReader br;
+
+//        Timeline tl = new Timeline(order);
+        Post tl = null;
+        this.url = this.baseUrl + this.midUrl + this.query;
+
+        try {
+            connection = new ClientConnection(this.url);
+        
+            try {
+                // get instance of bufferReader
+                br = getHtml(connection);
+                tl = scrape(br);
+            } catch (Exception e) {
+                DAO.trace(e);
+            } finally {
+                connection.close();
+            }
+        } catch (IOException e) {
+            DAO.severe("check internet connection");
+        }
+        return tl;
+    }
+
+    protected BufferedReader getHtml(ClientConnection connection) {
```

Comment:
Because of it's methods structure, one can't use only this method, it is dependent on other methods in the module. So it is necessary to inherit in the classes where needed to be used.